orttraining/orttraining/test/python/orttraining_test_experimental_gradient_graph.py

```diff
@@ -0,0 +1,125 @@
+import os
+import unittest
+from pathlib import Path
+
+import numpy as np
+import onnx
+import onnxruntime
+import torch
+from onnxruntime.training.experimental import export_gradient_graph
+
+
+class NeuralNet(torch.nn.Module):
+    r"""
+    Simple example model.
+    """
+
+    def __init__(self,
+                 input_size: int,
+                 embedding_size: int,
+                 hidden_size: int,
+                 num_classes: int):
+        super(NeuralNet, self).__init__()
+
+        self.frozen_layer = torch.nn.Linear(
+            input_size, embedding_size, bias=False)
+        # Freeze a layer (mainly to test that gradients don't get output for it).
+        self.frozen_layer.requires_grad_(False)
+
+        self.fc1 = torch.nn.Linear(embedding_size, hidden_size)
+        self.relu = torch.nn.ReLU()
+        self.fc2 = torch.nn.Linear(hidden_size, num_classes)
+
+    def forward(self, x):
+        out = self.frozen_layer(x)
+        out = self.fc1(out)
+        out = self.relu(out)
+        out = self.fc2(out)
+        return out
+
+
+def to_numpy(tensor):
+    return tensor.detach().cpu().numpy() if tensor.requires_grad else tensor.cpu().numpy()
+
+
+def binary_cross_entropy_loss(inp, target):
+    loss = -torch.sum(target * torch.log2(inp[:, 0]) +
+                      (1-target) * torch.log2(inp[:, 1]))
+    return loss
+
+
+class GradientGraphBuilderTest(unittest.TestCase):
+    def test_save(self):
```

Comment:
We could probably change the name of this test. I think let's address this separately later on.

Reply:
I don't might changing it but it's testing the `save` method of `GradientGraphBuilder` so it makes sense when you look at the test class name.

Reply:
I feel the test is doing more than just saving (it actually executes the `InferenceSession`). I think it is ok to leave it as `save` for now.